docs/topics/js/js-ir-compiler.md

```diff
@@ -36,6 +36,49 @@ kotlin {
 The compiler type can also be set in the `gradle.properties` file, with the key `kotlin.js.compiler=ir`.
 This behaviour is overwritten by any settings in the `build.gradle(.kts)`, however.
 
+## Lazy initialization of top-level properties
+
+For better application startup performance, the Kotlin/JS IR compiler initializes top-level properties lazily. This way,
+the application loads without initializing all the top-level properties used in its code. It initializes
+only the ones needed at startup; other properties receive their values later when the code that uses them actually runs.
+
+```kotlin
+val a = run { 
+    val result = // intensive computations
+    println(result)
+    result 
+} // value is computed upon the first usage
+```
+
+If for some reason you need to initialize a property eagerly (upon the application start), mark it with the 
+[`@EagerInitialization`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.js/-eager-initialization/){nullable="true"} annotation.
+
+## Incremental compilation for development binaries
+
+The JS IR compiler provides the _incremental compilation mode for development binaries_ that speeds up the development process.
+In this mode, the compiler caches the results of `compileDevelopmentExecutableKotlinJs` Gradle task on the module level.
+It uses the cached compilation results for unchanged source files during subsequent compilations, making them complete faster,
+especially with small changes.
+
+To enable incremental compilation for development binaries, add the following line to the project’s `gradle.properties`:
```

Comment:
it also could be added to `local.properties` and I think it's more popular in this case